tests/test_edge_cases.py

```diff
@@ -2346,3 +2346,23 @@ class Square(AbstractSquare):
         ),
     ):
         Square(side=1.0)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason='cannot use list.__class_getitem__ before 3.9')
+def test_generic_wrapped_forwardref():
+    class Operation(BaseModel):
```

Comment:
Without the changes in this PR, you actually get `{'type': 'PathItem'}` as an intermediate output of `GenerateSchema._generate_schema` (i.e., as an attempt at building a CoreSchema), which unsurprisingly makes class creation fail.